hoverkite-protocol/src/response.rs

```diff
@@ -31,14 +30,41 @@ pub enum Report {
     ChargeState {
         charger_connected: bool,
     },
+    PowerOff,
 }
 
-impl Report {
+impl Response {
+    pub fn log_from_fmt(args: core::fmt::Arguments<'_>) -> Self {
+        let mut writer = TruncatingWriter(ArrayString::new());
+
+        match core::fmt::Write::write_fmt(&mut writer, args) {
+            Ok(()) => (),
+            Err(_) => {
+                // `core::fmt::Error` doesn't have a payload, so we just have to guess.
```

Comment:
Maybe it would be easier to do this in `TruncatingWriter::write_str` instead? It should know for sure whether it needs to truncate, so it can add the ellipses and return `Ok`.

Reply:
No, because the fmt machinery will call write_str multiple times and we only want to do the ... dance once.

Reply:
actually I realised that I can override Write::write_fmt() so I have moved it in there, and it looks a bit neater.